--- a/drivers/mtd/Makefile		2015-06-26 21:07:17.000000000 +0800
+++ b/drivers/mtd/Makefile	2015-06-26 21:07:24.000000000 +0800
@@ -29,6 +29,7 @@
 obj-$(CONFIG_SM_FTL)		+= sm_ftl.o
 obj-$(CONFIG_MTD_OOPS)		+= mtdoops.o
 obj-$(CONFIG_MTD_SWAP)		+= mtdswap.o
+obj-$(CONFIG_SPI_MT7621)	+= ralink/
 
 nftl-objs		:= nftlcore.o nftlmount.o
 inftl-objs		:= inftlcore.o inftlmount.o

--- a/drivers/spi/Makefile		2015-06-26 09:42:38.077904289 +0800
+++ b/drivers/spi/Makefile	2015-06-26 21:03:46.000000000 +0800
@@ -46,7 +46,7 @@
 obj-$(CONFIG_SPI_MPC512x_PSC)		+= spi-mpc512x-psc.o
 obj-$(CONFIG_SPI_MPC52xx_PSC)		+= spi-mpc52xx-psc.o
 obj-$(CONFIG_SPI_MPC52xx)		+= spi-mpc52xx.o
-obj-$(CONFIG_SPI_MT7621)		+= spi-mt7621.o
+#obj-$(CONFIG_SPI_MT7621)		+= spi-mt7621.o
 obj-$(CONFIG_SPI_MXS)			+= spi-mxs.o
 obj-$(CONFIG_SPI_NUC900)		+= spi-nuc900.o
 obj-$(CONFIG_SPI_OC_TINY)		+= spi-oc-tiny.o
